packages/path_provider/path_provider_windows/lib/src/path_provider_windows_real.dart

```diff
@@ -262,4 +262,59 @@ class PathProviderWindows extends PathProviderPlatform {
     }
     return directory.path;
   }
+
+  @override
+  Future<List<String>?> getExternalStoragePaths(
+      {StorageDirectory? type}) async {
+    if (type == null) {
+      return _getLogicalDrives();
```

Comment:
This doesn't seem to correspond to the Android implementation, which is the de-facto standard for the behavior of the method (since it has previously been Android-only). From the Android docs:

> Returns absolute paths to application-specific directories on all shared/external storage devices where the application can place persistent files it owns. These files are internal to the application, and not typically visible to the user as media.

This does not describe a top-level drive.

> Shared storage devices returned here are considered a stable part of the device, including physical media slots under a protective cover. The returned paths do not include transient devices, such as USB flash drives connected to handheld devices.

This does not appear to be true of the implementation here.

Reply:
I assumed getExternalStorageDirectories() is to list storage drive roots to directly list users Storage. Last time I checked 
by default this function returns Phone and SD card roots. With proper permission this allows to list SD Card and Phone storage.  According to.
 https://github.com/flutter/packages/blob/d2705fb82b2605633e8528c053148128cc202327/packages/path_provider/path_provider/lib/path_provider.dart#L186-L188

So to reflect that here getLogicalDrive is being used since default root directory is a C drive. User may opt to add custom partiion to their existing drive or may have multiple drives internally insided the machine or a SD card slot similarly to Phones' SIM/SD card slot.

Reply:
If the androids behavior I have seen is unexpected then getLogicalDrive has to be removed